### PR TITLE
Using algolia for search functionality on site

### PR DESCRIPTION
**Pull request by Kev-Y-Huang**

### CODE REVIEW

**Comment by Kev-Y-Huang** [comment]
Hold off on accepting till I fix some of the netlify functions stuff

File: web/functions/deploy-building/deploy-building.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+// Sanity configuration
+const projectId = "xx0obpjv";
+const dataset = "production";
+
+exports.handler = function (event, context, cb) {
+  const URL = `https://${projectId}.api.sanity.io/v1/data/query/${dataset}/?query=*[_type == "project"]`;
+  console.log("Constructed URL is ...", URL);
+
+  // Initiate an Algolia client
+  const client = algoliasearch(algoliaApp, process.env.ALGOLIA_TOKEN);
```
Comment (mwjkwan) [inline web/functions/deploy-building/deploy-building.js:27]: Should this be added to Netlify?

File: web/src/pages/search.js [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+const ConnectedSearchBox = connectWithQuery(MySearchBox);
+
+const Hits = ({ hits }) => (
+  <Grid gap={4} columns={1}>
+    {hits.map((hit) => (
+      <ArticlePreview
+        horizontal={true}
+        link={hit.slug.current}
+        _rawExcerpt={hit.excerpt}
+        {...hit}
```
Comment (mwjkwan) [inline web/src/pages/search.js:201]: Awesome!

**Review by mwjkwan — APPROVED** [review]
Looks great! As search gets more complicated, we might want to split some of this functionality into separate files.

**Pull request merged**